File: crowdin/src/main/java/com/crowdin/platform/repository/local/LocalStringRepositoryFactory.java
package com.crowdin.platform.repository.local;

import android.content.Context;

import com.crowdin.platform.CrowdinConfig;

public class LocalStringRepositoryFactory {

    private LocalStringRepositoryFactory() {
    }

    public static LocalRepository createLocalRepository(Context context, CrowdinConfig config) {
        LocalRepository localRepository;
        if (config.isPersist()) {
            localRepository = new SharedPrefLocalRepository(context);
        } else {
            localRepository = new MemoryLocalRepository();
        }

        return localRepository;
    }
}

File: crowdin/src/main/java/com/crowdin/platform/repository/local/LocalStringRepositoryFactory.kt
package com.crowdin.platform.repository.local

import android.content.Context

import com.crowdin.platform.CrowdinConfig

internal object LocalStringRepositoryFactory {

    internal fun createLocalRepository(context: Context, config: CrowdinConfig): LocalRepository =
            when {
                config.isPersist -> SharedPrefLocalRepository(context)
                else -> MemoryLocalRepository()
            }
}
